logistical.js

```diff
@@ -7,23 +7,83 @@
  * Made available via the MIT License, text included in LICENSE.
  */
 
-var math = require('mathjs');
+var Validator = require('./lib/linear_support.js');
 
 /*
  * Main classifier entity.
  */
 var Classifier = function() {};
 
-/* Computes the logistic function value for a given input. */
+/*
+ * Intermediate function for calculating the partial sum for the likelihood
+ *
+ * w = vector coefficients with k elements
+ * Xi = vector of training data with k elements
+ */
+Classifier.prototype.ZiPartialSum = function(w, Xi) {
+  var validator = new Validator();
+
+  validator.isVector(w, Xi);
+
+  var N = validator.hasEqualElementCount(w, Xi);
+
+  var sum = 0;
+
+  /*
+   * @refactor
+   * Shouldn't this just be the dot product of the two vectors?
+   */
+  for ( var k = 1; k <= N; k++) {
```

Comment:
I have another branch where I refactored for the dot product so we can take this merge and then the next one will have that coming